src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java

```diff
@@ -672,6 +672,8 @@ public static void main(String[] args) {
     // Shutdown and cleanup connector
     shutDown();
     cleanUp();
+    // System.exit() is required for the event handler thread to stop running
+    System.exit(0);
```

Comment:
This is what I thought. I wanted to interrupt and join the event handler thread. However, the ETK does not provide this ( or I could not find it).
This is sort of a hack solution. But I don't like the call in cleanup.  If anyone ever added a function after cleanup it would never run.

Reply:
But its not obvious, or nowhere is it stated that nothing should be called after cleanUp. For example, the AbstractConnectorMain has a `postCleanUp` method that is called after. 
![image](https://github.com/user-attachments/assets/c611f9a6-9833-495b-970e-ce14c7b4bd9a)

Reply:
It could be clarified better that nothing is intended to go after cleanup. The intention of `shutDown()` and `cleanUp()`, though, was to isolate any of the shutdown and cleanup logic to a location outside of the main method (to avoid a long, cluttered main method).

Regarding `AbstractConnectorMain`, the same reasoning still stands, although the abstract connector framework was developed more recently, and the `postCleanUp()` method was an addition to help facilitate a cleaner connector/device restart procedure.

Ultimately, when time permitted, my goal was to migrate this Cumulocity connector to use the abstract connector framework. However, although this connector is similar to the abstract connector framework, there are minor differences and ambiguities, and the framework should be considered the 'better one' structure-wise.